pkg/k8s/watchers/pod.go

```diff
@@ -276,7 +277,26 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		// PodIPs update
 		fallthrough
 	case labelsChanged:
-		k.redirectPolicyManager.OnUpdatePod(newK8sPod)
+		lrpNeedsReassign = true
+	}
+
+	oldPodReady := slim_corev1.ConditionUnknown
+	newPodReady := slim_corev1.ConditionUnknown
+
+	for _, cond := range oldK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			oldPodReady = cond.Status
+			break
+		}
+	}
+	for _, cond := range newK8sPod.Status.Conditions {
+		if cond.Type == slim_corev1.PodReady {
+			newPodReady = cond.Status
+			break
+		}
+	}
```

Comment:
Can define a helper function to do this? `getLatestPodCondition` or something like that. And then you could reuse it in `(rpm *Manager) checkPodReady`

Reply:
Done.